Automattic/jetpack
# Search: add search setting page with only toggles

#### Changes proposed in this Pull Request:

Added search setting page, including Search and Instant Search toggles and a mocked instant search dialog.

#### Jetpack product discussion
pcNPJE-7J#comment-161

#### Does this pull request change what data or activity we track or use?
New tracks will be added in separate PR.

#### Testing instructions:
- Go to a site with Jetpack Instant Search subscription (enter URL directly `/wp-admin/admin.php?page=jetpack-search&a8ctest`)
- Ensure `Search` sub menu is added under Jetpack
- Click the `Search` sub menu
- Ensure the page is accessible with both PC and mobile devices
- Ensure the toggles work as expected
![image](https://user-images.githubusercontent.com/1425433/123947963-1061ec80-d9f5-11eb-8380-19848bdf2584.png)


NOTE: 
- The functionality is hidden, unless a `a8ctest` param is detected in URL, e.g. `/wp-admin/admin.php?page=jetpack-search&a8ctest`.
- `class.jetpack-admin-page.php` is moved to `projects/plugins/jetpack/_inc/lib/admin-pages/`, because if we disabled search module, Jetpack wouldn't load `modules/search` anymore. And I think it makes more sense to be there.
- Just trying to make the PR as small as possible - footer, buttons, Legacy search mock and tracking will be added in the follow-up PRs.
